core/model/modx/processors/context/getlist.class.php

```diff
@@ -60,6 +60,7 @@ public function prepareQueryBeforeCount(xPDOQuery $c) {
     public function prepareRow(xPDOObject $object) {
         $contextArray = $object->toArray();
         $contextArray['perm'] = array();
+        $contextArray['name'] = $object->get('name') != '' ? $object->get('name') : $object->get('key');
```

Comment:
I'd prefer if this was done in the combo element rather than in the list processor, as this affects other UI elements that use this processor which could confuse users if the `name` field which is empty now shows the same value as the `key`.

Reply:
Since it is a bit difficult to change the displayField in a combo during rendering, you should add a new array key there (i.e. `display_name`) and fill it with those values:

```
$contextArray['display_name'] = $object->get('name') != '' ? $object->get('name') : $object->get('key');
```

Afterwards you could refer that key in the combo (change `name` to `display_name`):
https://github.com/modxcms/revolution/blob/05d23c0478fb7cf26a5921ebd9a31d2c7fbfccfc/manager/assets/modext/widgets/core/modx.combo.js#L336-L338

Reply:
@opengeek I do not agree with you? So if you have 3 different combo boxes (maybe people make a own combo box instead of the context combo box) you will still have empty rows.

The getlist processor is the source of the 'data' so you need to change it in the source, not on 10 different places in 10 different UI elements... In a normal situation the name is set, if not it has a fallback on the key and the key will be the name of the context...

Reply:
It is never a good idea to use the default field key and change the value. That key/value is posted back i.e. during inline editing and could lead to unwanted values in that table.

This was the reason, why it was removed in the processor. See https://github.com/modxcms/revolution/issues/13371 and https://github.com/modxcms/revolution/commit/62a3d33348661cb46836b20f4cc69c132ad6cd0a

Reply:
@Jako thats is a different issue, it ADDS something to a name instead of a fallback, and yes then you will get unwanted values... The name of a context should be required, and a fallback for the name is needed because a MODx update.

But I won't fix this in ExtJS because thats not the good solution.

Reply:
It is not a very different issue. Both could change the original value, which should not happen during a getlist (even if the new value 'should' be right).